HotSpot/View Controllers/MainContainerViewController.m

```diff
@@ -86,49 +119,38 @@ - (IBAction)switchMode:(id)sender {
         self.mapView.hidden = NO;
         self.spotListView.hidden = YES;
     }
+    
+    // cute little animation that is unnecessary but necessary
+    // expands and then shrinks
+    [UIView animateWithDuration:.2
+                     animations:^{
+                         self.modeSwitchButton.transform = CGAffineTransformMakeScale(1.5, 1.5);
+                     }completion:^(BOOL finished) {
+                         [UIView animateWithDuration:.35 animations:^{
+                             self.modeSwitchButton.transform = CGAffineTransformIdentity;
+                         }];
+                     }];
 }
 
 # pragma mark - Search Related
 
 - (void)searchBar:(UISearchBar *)searchBar textDidChange:(NSString *)searchText{
     
-    // setting the views to hidden or not
-    self.searchResultTableView.hidden = NO;
-    
-    // starting to try out a simple animation 
-    if(self.searchResultTableView.frame.size.height == 0){
+    // this is the animation for a search results drop down
+    if(self.searchResultTableView.frame.size.height ==0)
+        [UIView animateWithDuration:0.4 delay:0.0 options:UIViewAnimationOptionCurveEaseIn animations:^{CGRect frame = self.searchResultTableView.frame;
+        // set to size of the view controller
+            frame.size.height = 800;
```

Comment:
as discussed, you should never use constants for things like height, since the height is different for different screen sizes. Also, PLEASE TEST ON ANOTHER DEVICE SIZES! Simulator makes this really easy.

Reply:
I didn't work at all on animations last night so this will be updated